src/middlewared/middlewared/etc_files/netdata/netdata.conf.mako

```diff
@@ -1,13 +1,11 @@
 <%
-	from middlewared.plugins.reporting.netdata.utils import NETDATA_PORT, NETDATA_UPDATE_EVERY
-
-
-	if not middleware.call_sync('reporting.netdata_setup'):
-		# Let's exit if netdata storage is not in place
-		middleware.logger.error('Netdata configuration file could not be generated')
-		raise FileShouldNotExist()
-
-	disk_space = middleware.call_sync('netdata.get_disk_space')
+    from middlewared.plugins.reporting.netdata.utils import NETDATA_PORT, NETDATA_UPDATE_EVERY
+    netdata_cache_dataset = middleware.call_sync('reporting.netdata_storage_location')
+    disk_space = middleware.call_sync('netdata.get_disk_space')
```

Comment:
Is there a reason you're populating `disk_space` before the `if not netdata_cache_dataset` check with these changes? Seems like we should revert back to what you were doing before these changes. (i.e. only setting that variable after that check)